DDD-17 Setup initial entities

KornelisMax opened the pull request:
Created entities and associations between them according to our entity [diagram](https://ba-jira.atlassian.net/jira/software/projects/DDD/boards/5?selectedIssue=DDD-17) (file attached in jira ticket).
In addition, created one to many relationship between user and order (we are planning to add functionality to see order history for registered user). For now, I added `CASCADE ALL` actions only to `user`, `user_info`, `address` fields, because only these fields are going to be edited together (but probably this needs more investigation). 

JIRA task: https://ba-jira.atlassian.net/jira/software/projects/DDD/boards/5?selectedIssue=DDD-17

@egisxxegis @Gintasp @Accir @Zygis0321

KornelisMax committed:
Create initial entities

KornelisMax committed:
DDD-17 Change update to createe.xml

KornelisMax committed:
DDD-17 Add relationships to entities and asign table names manually

KornelisMax committed:
DDD-17 Add hashcodes and equal overrides to entities

KornelisMax commented on src/main/resources/META-INF/persistence.xml:
```diff
@@ -6,7 +6,7 @@
         <properties>
             <property name="hibernate.show_sql" value="true"/>
-            <property name="hibernate.hbm2ddl.auto" value="update"/>
+            <property name="hibernate.hbm2ddl.auto" value="create"/>
```
After first launch don't forget to change it back to `value="update"`

giplonis replied:
I don't know if we even need to use `create` as `update` seems to do the same thing 🤔

KornelisMax commented on src/main/java/lt/vu/persistence/orm/entities/Address.java:
```diff
@@ -0,0 +1,50 @@
+    private User user;
+
+    @OneToOne
+    private UserInfo guest;
```
We should decide on registered/guest user naming. I think guest fits quite well here.

giplonis committed:
DDD-17 Make relationships unidirectional, add additional entities

Added PackageType entity (we will need to have some titles and descriptions for those types). Removed bidirectional relationships as they cause consistency issues. Removed unnecessary relations. Renamed Package to more sensible PackageOption.

Gintasp commented:
Nice! To move things quicker I also pushed some improvements while reviewing:
- Removed bidirecational relationships (they generally cause data consistency issues in application logic level)
- Removed unnecessary relationships (for example, Package does not need to know anything about Order)
- `OneToOne` is rarely needed and it's usually a sign of bad design. Same can be achieved using `ManyToOne`. More info [here](https://stackoverflow.com/questions/2452987/hibernate-why-use-many-to-one-to-represent-a-one-to-one).
- Added a couple of new entities cause I realized we'll need some extra data
- Rename `Package` to more sensible `PackageOption`
